crates/bevy-inspector-egui/src/bevy_inspector/mod.rs

```diff
@@ -292,31 +307,83 @@ impl Filter {
     }
 
     /// filter entities based on internal state
-    fn filter_entities(&self, world: &mut World, entities: &mut Vec<Entity>) {
-        if self.word.is_empty() {
-            return;
+    fn _filtered_entities(
+        &self,
+        world: &mut World,
+        entities: impl IntoIterator<Item = Entity>,
+    ) -> Vec<Entity> {
+        let entities = entities.into_iter();
+        if self.is_empty() {
+            entities.collect()
+        } else {
+            entities
+                .filter(|&entity| {
+                    self_or_children_satisfy_filter(
+                        world,
+                        entity,
+                        self.word.as_str(),
+                        self.is_fuzzy,
+                    )
+                })
+                .collect()
         }
+    }
+}
 
-        entities.retain(|entity| {
-            self_or_children_satisfy_filter(world, *entity, self.word.as_str(), self.is_fuzzy)
-        });
+impl EntityFilter for Filter {
+    fn is_empty(&self) -> bool {
+        self.word.is_empty()
+    }
+    fn filtered_entities(
+        &self,
+        world: &mut World,
+        entities: impl IntoIterator<Item = Entity>,
+    ) -> Vec<Entity> {
```

Comment:
I'm tempted to revert this to the original signature `filter_entities`. It is part remnant of some more "complex" implementation I was working on and part consideration for the implementor having more flexibility, potentially at the expense of the extra overhead even though Entity is Copy.

Reply:
Yeah, I think it makes more sense to [revert it like this](https://github.com/jakobhellermann/bevy-inspector-egui/pull/231/commits/c8010db3e636fc25c4431d0469fdc8434a0ab13a). I believe the `.collect()` was making the `IntoIterator` param useless since we couldn't take advantage of lazy iteration and the user isn't the one providing the `Vec<Entity>` anyway.